src/main/kotlin/org/neo4j/graphql/Translator.kt

```diff
@@ -359,75 +359,79 @@ class Translator(val schema: GraphQLSchema) {
-                        GraphQLArgument("from", "from field name", Scalars.GraphQLString, "from"),
-                        GraphQLArgument("to", "to field name", Scalars.GraphQLString, "to")), false, false, true))
-        return directives
-    }
-
     private fun augmentSchema(typeDefinitionRegistry: TypeDefinitionRegistry, schemaParser: SchemaParser, ctx: Translator.Context): TypeDefinitionRegistry {
+        val directivesSdl = """
+            enum RelationDirection {
+               IN
+               OUT
+            }
+            directive @relation(name:String, direction: RelationDirection = OUT, from: String = "from", to: String = "to") on FIELD_DEFINITION | OBJECT
+            directive @cypher(statement:String) on FIELD_DEFINITION
+            directive @property(name:String) on FIELD_DEFINITION
+        """
+        typeDefinitionRegistry.merge(schemaParser.parse(directivesSdl))
 
         val augmentations = typeDefinitionRegistry.types().values.filterIsInstance<ObjectTypeDefinition>().map { augmentedSchema(ctx, it) }
 
         val augmentedTypesSdl = augmentations.flatMap { listOf(it.filterType, it.ordering, it.inputType).filter { it.isNotBlank() } }.joinToString("\n")
         typeDefinitionRegistry.merge(schemaParser.parse(augmentedTypesSdl))
 
-        val schemaDefinition = typeDefinitionRegistry.schemaDefinition().orElseGet { SchemaDefinition() }
-        if (!typeDefinitionRegistry.schemaDefinition().isPresent) typeDefinitionRegistry.add(schemaDefinition).ifPresent { throw RuntimeException(it.toSpecification().toString()) }
+        val schemaDefinition = typeDefinitionRegistry.schemaDefinition().orElseGet { SchemaDefinition.newSchemaDefinition().build() }
+        val operations = schemaDefinition.operationTypeDefinitions
+                .associate { it.name to typeDefinitionRegistry.getType(it.typeName).orElseThrow({ RuntimeException("Could not find type: " + it.typeName) }) as ObjectTypeDefinition }
+                .toMap()
 
-        val operations = schemaDefinition.operationTypeDefinitions.associate { it.name to typeDefinitionRegistry.getType(it.type).get() as ObjectTypeDefinition }.toMap()
 
         val queryDefinition = operations.getOrElse("query") {
-            ObjectTypeDefinition("Query").also {
-                schemaDefinition.operationTypeDefinitions.add(OperationTypeDefinition("query", TypeName(it.name)))
-                typeDefinitionRegistry.add(it)
-            }
-        }
-        val mutationDefinition = operations.getOrElse("mutation") {
-                ObjectTypeDefinition("Mutation").also {
-                    schemaDefinition.operationTypeDefinitions.add(OperationTypeDefinition("mutation", TypeName(it.name)))
-                    typeDefinitionRegistry.add(it)
+            typeDefinitionRegistry.getType("Query", ObjectTypeDefinition::class.java).orElseGet {
+                ObjectTypeDefinition("Query").also { typeDefinitionRegistry.add(it) }
```

Comment:
it's not fully equivalent to the before, as before we did add it if it was not there?